.github/renovate.json

```diff
@@ -106,12 +106,45 @@
       "matchManagers": ["npm", "nvm"],
       "additionalBranchPrefix": "fe-"
     },
+    {
+      "description": "Exclude frontend deps until Optimize migrates to react-router v6 and react v18",
+      "matchManagers": ["npm", "nvm"],
+      "matchFileNames": ["optimize/**"],
+      "matchPackageNames": [
+        "@carbon/react",
+        "old-bpmn-js",
+        "react",
+        "react-dom",
+        "react-router",
+        "react-router-dom",
+        "@types/react",
+        "@types/react-dom",
+        "@types/react-router-dom"
+      ],
+      "enabled": false
+    },
+    {
+      "description": "Exclude Optimize backend dependencies",
+      "matchManagers": ["maven"],
+      "matchFileNames": ["optimize/**"],
+      "matchPackageNames": [
+        "io.camunda.optimize:optimize-data-generator",
+        "io.camunda.optimize:camunda-optimize",
+        "io.camunda.optimize:util",
+        "io.camunda:identity-sdk",
+        "io.camunda:identity-spring-boot-starter",
+        "io.camunda:zeebe-client-java",
+        "io.camunda:zeebe-protocol",
+        "ant-contrib:ant-contrib"
+      ],
+      "enabled": false
+    },
     {
       "matchPackageNames": ["mcr.microsoft.com/playwright"],
       "additionalBranchPrefix": "fe-"
     },
     {
-      "matchPackagePrefixes": ["@types/"],
+      "matchPackageNames": ["/^@types/"],
```

Comment:
in this file, this gets changed a lot. What is the reason for it?

Reply:
As mentioned in the PR description, I did that because of the [renovate-lint workflow](https://github.com/camunda/camunda/actions/runs/10354638209) was failing

Reply:
ah sorry, missed that 👍

Reply:
I tried this change before and had to revert due to [Renovate behaving differently afterwards](https://camunda.slack.com/archives/C071KP5BTHB/p1722856465936289?thread_ts=1722845758.260349&cid=C071KP5BTHB), in #20772.

Did anyone check now whether the config migration was successful and no unintended behavior happened after merging this PR? (since Renovate affects the whole monorepo)

Reply:
@cmur2 what do you mean?

Reply:
@misiekhardcore this PR #21060 migrates the Renovate configuration which was already done in #20694 and reverted in #20772. The reason for the revert was that Renovate opened PRs for dependencies that should be ignored but apparently were not anymore (maybe the config was wrong or there was a bug), see [Slack](https://camunda.slack.com/archives/C071KP5BTHB/p1722856465936289?thread_ts=1722845758.260349&cid=C071KP5BTHB).

Given the history of #20694 and the revert in #20772 my ask is to make sure that after this PR #21060 Renovate still works as expected. If not, we have to revert the config migration again.

Since Renovate configuration changes may impact the whole monorepo we have to check after these changes that Renovate still works as expected with the new configuration.

Reply:
I asked if this kind of check has been done.

Reply:
Just now @ThorbenLindhauer might have noticed a behavior that is related to this repeated configuration migration that is not intended, see [Slack](https://camunda.slack.com/archives/C071KP5BTHB/p1724147032048869).

Reply:
I see. But how can I check the renovate config in other way than relaying in the renovate validator? That was the reason of this migration - renovate config validation. If these changes are breaking things, should I open a PR to revert them?

Reply:
> That was the reason of this migration - renovate config validation.

No worries! I understand that since it happened the same to me, in the past the config validator has always been helpful but in this instance it seems to propose breaking changes for some reason which is unfortunate :(

> how can I check the renovate config in other way

The main thing that counts is the behavior of Renovate based on the config (kind of an E2E test), as we learned even the config validator migration can be wrong. When performing a config change which shouldn't change the behavior (such as a no-op config migration) key aspects to watch out for are:

* the list of detected dependencies (full list on https://developer.mend.io/github/camunda/camunda) should not change
* the list of open/to be opened PRs should not change

Looking at the git history of the `.github/renovate.json` can also provide insights about what was tried recently.

> should I open a PR to revert them

Yes, please 👍🏼 As far as I see we need to only revert https://github.com/camunda/camunda/pull/21060/commits/7ff63cd2592a39b4c8a8b22441211ee45569f764, right?

Reply:
okay, I will open a PR and assign you as a reviewer

> the list of detected dependencies (full list on https://developer.mend.io/github/camunda/camunda) should not change

I don't have access to this page

> the list of open/to be opened PRs should not change

this can be tricky, as renovate is set to work "at any time" so it can open and merge PRs at any given point of time, so looking at the list of deps PRs before and after (if of course this list didn't blow from 10 to 100 PRs) won't say much

Reply:
here is the [PR](https://github.com/camunda/camunda/pull/21367)

Reply:
> I don't have access to this page

You should if you login to that site via GitHub since I think it needs no special permissions 🤔 what error do you get?

Reply:
![Screenshot from 2024-08-20 14-05-00](https://github.com/user-attachments/assets/afad8012-98ca-4d8f-b168-6cc83aad3d3e)

Reply:
and when I go to https://developer.mend.io/github/camunda then I get
![Screenshot from 2024-08-20 14-06-04](https://github.com/user-attachments/assets/0898bfbc-e19a-4805-b364-558c8bad7740)

Reply:
Fixed this via DMs, access is now granted :)